inst/include/population_dynamics/population/population.hpp

```diff
@@ -33,15 +33,366 @@
+          
+            
+            //size all the vectors to length of nages
+            nfleets = fleets.size();
+            ages.resize(nages);
+            catch_at_age.resize(nages);
+            catch_numbers_at_age.resize((nyears+1)*nages*nfleets);
+            mortality_F.resize(nyears*nages);
+            mortality_Z.resize(nages);
+            proportion_mature_at_age.resize((nyears+1)*nages);
+            initial_numbers.resize(nages);
+            weight_at_age.resize(nages);
+            catch_weight_at_age.resize(nyears*nages*nfleets);
+            unfished_numbers_at_age.resize((nyears+1)*nages);
+            numbers_at_age.resize((nyears+1)*nages);
+            expected_catch.resize(nyears*nfleets);
+            expected_index.resize(nyears*nfleets);
+            biomass.resize((nyears+1));
+            unfished_spawning_biomass.resize((nyears+1));
+            spawning_biomass.resize((nyears+1));
+            log_naa.resize(nages);
+            log_Fmort.resize(nfleets*nyears);
+            log_M.resize(nyears*nages);
+            log_q.resize(nfleets);
+            naa.resize(nages);
+            Fmort.resize(nfleets*nyears);
+            M.resize(nyears*nages);
+            q.resize(nfleets);
+
+            
+            
+        }
+
+        //gets called at each model iteration (used to zero out derived quantities, values summed with +=, etc.)
+        void Prepare() {
+          std::fill(unfished_spawning_biomass.begin(), unfished_spawning_biomass.end(), 0);
+          std::fill(spawning_biomass.begin(), spawning_biomass.end(), 0);
+          std::fill(mortality_F.begin(), mortality_F.end(), 0);
+          std::fill(expected_catch.begin(), expected_catch.end(), 0);
+
+          //Transformation Section
+          for (int age = 0; age <= this->nages; age++) {
+            this -> naa[age] = exp(this -> log_naa[age]);
+            for(int year = 0; year < this->nyears; year++) {
+              int index_ya = year*nages + age;
+              this -> M[index_ya] = exp(this -> log_M[index_ya]);
+            }
+          }
+
+          for(int fleet_ = 0; fleet_ <= this->nfleets; fleet_++) {
+            this -> Fmort[fleet_] = exp(this -> log_Fmort[fleet_]);
+            for(int year = 0; year < this->nyears; year++) {
+              int index_yf = year * nfleets + fleet_;
+              this -> q[index_yf] = exp(this -> log_q[index_yf]);
+            }
+          }
+          // call functions to set up recruitment deviations.
+          this -> recruitment -> PrepareConstrainedDeviations();
+          this -> recruitment.PrepareBiasAdjustment();
+        }
+
+        /**
+         * life history calculations
+         */
+        //delete std::vector<Type> because already declared; function has access to log_naa
+        //exp(this ->log_naa[a]) on line 130 - more explicit that we are reference member
+        inline void CalculateInitialNumbersAA(int index, int a) { //inline all function unless complicated
```

Comment:
Should we rename this to SetInitialNumbersAtAge?

Reply:
We just had Calculate because it was consistent with all the other functions. I aggree though this is just setting the value not calculating anything. I don't mind either way what name we choose.